### PR TITLE
Support to INSPECT & DEID parquet files

### PR DESCRIPTION
**Pull request by chitara-01**
<h4> Summary (Short summary of what is being done) : </h4>
Added support to INSPECT & DEID parquet files from GCS bucket and store results in BQ datasets.
<br>
<h4> Description (Describe in detail the fix made) : </h4>
Introducing a dedicated java package to read the data from parquet files as GenericRecord objects, flatten each record and convert to Table.Row objects for further processing. This change works for inspection and de-identification of input files stored in GCS storage buckets. The results are written in BigQuery datasets. The tables from the BigQuery datasets can be further re-identified in the usual manner. 
<br>
<h4> Bug ID (if any) : </h4>
b/293426633
<br>
<h4> Public Documentation (if any) : </h4>
<br>
<h4> TESTED (Test Cases with scenario and description - must have 1 positive and 1 negative scenario) : </h4>
1. Converted CCRecords sample data from CSV format to Parquet format and tested both inspection and de-identification pipelines.
<br>
2. Tested on parquet data from [this github repository](https://github.com/kaysush/sample-parquet-files) 
<br>
3. Tested on various other parquet sample data from sources like kaggle, github, etc.
<br>
4. Introduced CI testing for Parquet pipeline - both inspection and de-identification.
<br>
5. Updated with PR suggestions/corrections and tested.

### CODE REVIEW

File: src/main/java/com/google/swarm/tokenization/parquet/ByteValueConverter.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 Google LLC
```
Comment (Goutam1511) [inline src/main/java/com/google/swarm/tokenization/parquet/ByteValueConverter.java:2]: 2023

File: src/main/java/com/google/swarm/tokenization/parquet/GenericRecordFlattener.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2021 Google LLC
```
Comment (Goutam1511) [inline src/main/java/com/google/swarm/tokenization/parquet/GenericRecordFlattener.java:2]: 2023

File: .github/workflows/dlp-pipelines.yml [hunk 3]
```diff
@@ -9,17 +9,23 @@ on:
 env:
   PROJECT_ID: "dlp-dataflow-deid-ci-392604"
   DATASET_ID: "demo_dataset"
+  PARQUET_DATASET_ID: "parquet_results"
```
Comment (Goutam1511) [inline .github/workflows/dlp-pipelines.yml:12]: Why do we need a new dataset and separate folder to keep the parquet files?

File: src/main/java/com/google/swarm/tokenization/parquet/GenericRecordFlattener.java [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+         */
+        private void processArray(Object value, Schema fieldSchema, String fieldKey) {
+            List<?> array = (List<?>) value;
+            for (int index = 0; index < array.size(); index++) {
+                processType(array.get(index),
+                            fieldSchema.getElementType(),
+                            String.format("%s[%s]", fieldKey, index),
+                            null);
+            }
+        }
```
Comment (saurabhgadia4) [inline src/main/java/com/google/swarm/tokenization/parquet/GenericRecordFlattener.java:207]: This is not getting used. Please remove

File: src/main/java/com/google/swarm/tokenization/parquet/ParquetReaderSplittableDoFn.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+            SeekableByteChannel seekableByteChannel = readableFile.openSeekable();
+            AvroParquetReader.Builder builder = AvroParquetReader.<GenericRecord>builder(
+                    new BeamParquetInputFile(seekableByteChannel));
+
+            try(ParquetReader<GenericRecord> fileReader = builder.withFileRange(start, end).build()) {
+                GenericRecord record;
+                while ((record = fileReader.read()) != null) {
+                    Table.Row tableRow = RecordFlattener.forGenericRecord().flatten(record);
+                    if (fileReader.getCurrentRowIndex() == 0)
+                        LOG.info("0th index: {}", tableRow.getValuesList());
```
Comment (dup05) [inline src/main/java/com/google/swarm/tokenization/parquet/ParquetReaderSplittableDoFn.java:70]: We should avoid logging the data.

File: src/main/java/com/google/swarm/tokenization/parquet/ParquetColumnNamesDoFn.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        String fileName = c.element().getKey();
+        ReadableFile parquetFile = c.element().getValue();
+
+        SeekableByteChannel seekableByteChannel = parquetFile.openSeekable();
+        AvroParquetReader.Builder builder = AvroParquetReader.<GenericRecord>builder(new BeamParquetInputFile(seekableByteChannel));
+
+        try (ParquetReader<GenericRecord> fileReader = builder.build()) {
+            GenericRecord record = fileReader.read();
+            List<String> flattenedFieldNames = RecordFlattener.forGenericRecord().flattenColumns(record);
+            LOG.info("Column names after flattening: ", flattenedFieldNames);
```
Comment (saurabhgadia4) [inline src/main/java/com/google/swarm/tokenization/parquet/ParquetColumnNamesDoFn.java:34]: If you think this log will be helpful in debugging then its worth keeping it. Else might want to remove it or consider LOG.Debug().

**Review by saurabhgadia4 — APPROVED** [review]

File: src/main/java/com/google/swarm/tokenization/parquet/ParquetColumnNamesDoFn.java (hunk 6 above)
Comment (chitara-01) [reply src/main/java/com/google/swarm/tokenization/parquet/ParquetColumnNamesDoFn.java]: Removed the statement.

**Review by Goutam1511 — APPROVED** [review]
LGTM with new changes

**Pull request merged**